### PR TITLE
{test} Set up basic testing with GoogleTest

### PR DESCRIPTION
**Pull request by asmaloney**
This sets up a testing framework using [GoogleTest](https://google.github.io/googletest/).

It uses test data from another repo ([libE57Format-test-data](https://github.com/asmaloney/libE57Format-test-data)) to keep this one a reasonable size. If the test data is not found, it will still run a coupe of basic write tests.

It is set up to run in the CI with the test data from the other repo.

I am looking for feedback on this setup before I merge it.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -186,6 +186,15 @@ install(
...
+# Testing
+option( E57_BUILD_TEST "Build tests" ${E57_BUILDING_SELF} )
+
+if ( E57_BUILD_TEST )
+    enable_testing()
+
+    add_subdirectory( test )
+endif()
+
 # CMake package files
```
Comment (nigels-com) [inline CMakeLists.txt:198]: Perhaps some preference _not_ to auto-enable testing.  (opt-in?)
We have some very large mono-repo style builds that would likely want to skip that step.

Comment (nigels-com) [reply CMakeLists.txt:198]: On the other hand I am sympathetic to the "test by default" school of thinking.

File: test/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# libE57Format Testing
+
+Testing uses the [googletest](https://github.com/google/googletest) framework. The documentation for it may be [found here](https://google.github.io/googletest/).
```
Comment (nigels-com) [inline test/README.md:3]: googletest is indeed one of the go-to solutions nowadays, widely used for the same purpose elsewhere.

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+## Turning Testing On
+
+To turn testing on, set the CMake option `E57_BUILD_TEST` to ON.
+
+## Testing Data
+
+Currently, the testing data is found in another repo: [libE57Format-test-data](https://github.com/asmaloney/libE57Format-test-data).
+
+As we build out testing, the number & size of the test data might grow very quickly. For this reason, the test data is kept separate from the main repo. This gives us some flexibility depending on how large the data set becomes. If ends up being too big for a git repo, for example, we can make it available as a zip download somewhere. It will also let us manage the data in CI better so it doesn't have to be downloaded every time CI is run.
+
```
Comment (nigels-com) [inline test/README.md:14]: Another advantage of a tarball approach is less git submodule and (possibly?) git LFS wrangling.  Test data doesn't tend to change from one commit to the next.

**Comment by nigels-com** [comment]
Overall does look like a very good start.  _Any_ testing is better than _none_, as I like to say.

File: CMakeLists.txt (hunk 1 above)
Comment (asmaloney) [reply CMakeLists.txt:198]: I was going for an "in-between" solution. If you are building libE57Format directly it should be enabled by default (`${E57_BUILDING_SELF}`). If it's included as a subdirectory, it should be disabled by default.

File: test/README.md (hunk 3 above)
Comment (asmaloney) [reply test/README.md:14]: Yes I agree. The drawback is managing the tarball hosting, etc.. Using a repo (for now) makes storage and tracking changes/providing an interface into it easy. I also like having it all available though one site (GitHub).

If we start using it heavily the data will grow beyond a git repo, however, which is why I set it up this way. Easy to switch it out.

(My upload speed is not so great, so managing multiple gigabytes is going to be painful. 😄)

File: CMakeLists.txt (hunk 1 above)
Comment (nigels-com) [reply CMakeLists.txt:198]: Yes, entirely reasonable.

**Pull request merged**